File: C/009.c
#include <stdio.h>

int solution()
{
	int c;
	for (int a = 3; a < 1000; a++){
		for (int b = a + 1; a + b < 1000; b++){
			c = b + 1;
			while (a*a + b*b > c*c) c++;
			if (a*a + b*b == c*c && a + b + c == 1000){
				printf("a:%d b:%d c:%d\n",a,b,c);return a*b*c;}
		}
	}
	return 0;
}


int main(void)
{
	
	printf("%d\n",solution());

	return 0;
}

File: Haskell/009.hs
import Data.Time

main :: IO ()
main = do 
  !startTime <- getCurrentTime
  let !result = solution
  !endTime <- getCurrentTime
  let diffTime = diffUTCTime endTime startTime
  putStrLn ("Solution: " ++ (show result))
  putStrLn ("Execution Time: " ++ (show diffTime))
  --putStrLn $ (reverse.tail.reverse) (show diffTime)

solution :: Int 
solution = head triplet

triplet :: [Int]
triplet = [a*b*c | a <- [1..998], b <-[1..998], let c = (1000-a-b), a^2 + b^2 == c^2]

